crates/sui-node/src/main.rs

```diff
@@ -49,6 +49,10 @@ fn main() {
     // TODO: re-enable after we figure out how to eliminate crashes in prod because of this.
     // ProtocolConfig::poison_get_for_min_version();
 
+    move_vm_profiler::gas_profiler_feature! {
+        error!("Cannot run the sui-node binary with gas-profiler feature enabled");
```

Comment:
I would recommend making this an actual compilation error (since this will be determined at that time). This can be done using `compile_error` https://doc.rust-lang.org/std/macro.compile_error.html

Reply:
https://github.com/MystenLabs/sui/pull/14592#discussion_r1422073761

Reply:
I tried this again with the macro update but got the same result, I think we need to leave it as a runtime error